io.home-assistant/index.js

```diff
@@ -0,0 +1,279 @@
+// -*- mode: js; indent-tabs-mode: nil; js-basic-offset: 4 -*-
+//
+// This file is part of io.home-assistant
+//
+// Copyright 2019 Giovanni Campagna <gcampagn@cs.stanford.edu>
+//
+// See LICENSE for details
+"use strict";
+
+const Tp = require('thingpedia');
+const WebSocket = require('ws');
+const HomeAssistant = require('home-assistant-js-websocket');
+
+// FIXME make configurable
+const HASS_URL = 'http://hassio.local:8123';
+
+class HomeAssistantDevice extends Tp.BaseDevice {
+    constructor(engine, state, master, entityId) {
+        super(engine, state);
+
+        this.master = master;
+        this._entityId = entityId;
+        this.uniqueId = master.uniqueId + '/' + entityId;
+        this.name = this.state.friendly_name;
+    }
+
+    updateState(state) {
+        super.updateState(state);
+        this.name = this.state.friendly_name;
+    }
+
+    _callService(domain, service, data = {}) {
+        data.entity_id = this._entityId;
+        return HomeAssistant.callService(this.master.connection, domain, service, data);
+    }
+}
+
+class HomeAssistantLightbulbDevice extends HomeAssistantDevice {
```

Comment:
There are a lot of domains, I would recommend to break these out into different files. It will also make it easier for people to contribute, since there will be less merge conflicts.